### PR TITLE
[openslide] add port

### PR DESCRIPTION
**Pull request by Neumann-A**
- [?] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md) (hoepfully)
- [x] [The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.](https://repology.org/project/openslide/versions)
- [ ] ~~Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)~~
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [ ] ~~The generated "usage text" is accurate. See docs/examples/adding-an-explicit-usage.md for context.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

probably blocked by #29258

### CODE REVIEW

**Comment by Neumann-A** [comment]
@BillyONeal  Why did you close this one exactly?

**Comment by BillyONeal** [comment]
> @BillyONeal Why did you close this one exactly?

The usual "github closed it when the other PR merged for some reason"

**Comment by BillyONeal** [comment]
Ah, I said 'resolves' there to indicate that it resolved the problem encountered here, but github interpreted that as 'this PR makes that other one obsolete'. Sorry about that!

File: ports/openslide/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,57 @@
+vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
+if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
+  set(PATCHES fix-win-build.patch)
+endif()
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO openslide/openslide
+    REF 4cd4b83c3ffff0382126d0618c85407f2c469b28
```
Comment (Cheney-W) [inline ports/openslide/portfile.cmake:9]:
```suggestion
    REF "v${VERSION}" 
```

File: ports/openslide/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+)
+
+vcpkg_install_make()
+vcpkg_fixup_pkgconfig()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+# Handle copyright
+file(INSTALL "${SOURCE_PATH}/LICENSE.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (Cheney-W) [inline ports/openslide/portfile.cmake:57]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE.txt")
```

**Review by Cheney-W — CHANGES_REQUESTED** [review]

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**